packages/@aws-cdk/aws-pipes-targets-alpha/test/integ.api-destination.ts

```diff
@@ -0,0 +1,64 @@
+import { IPipe, ISource, Pipe, SourceConfig } from '@aws-cdk/aws-pipes-alpha';
+import { IntegTest } from '@aws-cdk/integ-tests-alpha';
+import * as cdk from 'aws-cdk-lib';
+import { ApiDestinationTarget } from '../lib/api-destination';
+
+const app = new cdk.App();
+const stack = new cdk.Stack(app, 'aws-cdk-pipes-targets-api-dest');
+const sourceQueue = new cdk.aws_sqs.Queue(stack, 'SourceQueue');
+
+class TestSource implements ISource {
+  sourceArn: string;
+  sourceParameters = undefined;
+  constructor(private readonly queue: cdk.aws_sqs.Queue) {
+    this.queue = queue;
+    this.sourceArn = queue.queueArn;
+  }
+  bind(_pipe: IPipe): SourceConfig {
+    return {
+      sourceParameters: this.sourceParameters,
+    };
+  }
+  grantRead(pipeRole: cdk.aws_iam.IRole): void {
+    this.queue.grantConsumeMessages(pipeRole);
+  }
+}
+
+const secret = new cdk.aws_secretsmanager.Secret(stack, 'MySecret', {
+  secretStringValue: cdk.SecretValue.unsafePlainText('abc123'),
+});
+
+const connection = new cdk.aws_events.Connection(stack, 'MyConnection', {
+  authorization: cdk.aws_events.Authorization.apiKey('x-api-key', secret.secretValue),
```

Comment:
I'm also curious to see what happens when you also set the same header in `ApiDestination.headerParameters`. It's either ignored or overrides the `authorization`, but I think we should probably warn the user either way

Reply:
See comment below with screenshot.